lib/public/components/runEorReasons/runEorReasonSelection.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import { h } from '/js/src/index.js';
+
+/**
+ * A component with dropdowns allowing the user to select an EOR reason type
+ * @param {Object} model A model which can store the selected EOR reason
+ * @param {Object[]} eorReasonTypes the possible EOR reason types to display in the selection.
+ * @returns {vnode} A pair of dropdowns to set the eor reason category and type
+ */
+export const eorReasonSelectionComponent = (model, eorReasonTypes) => {
+    const reasonTypeCategories = [];
+
+    for (const { category } of eorReasonTypes) {
+        if (!reasonTypeCategories.includes(category)) {
+            reasonTypeCategories.push(category);
+        }
+    }
+
+    return [
+        h('.flex-row', [
+            h('select#eorCategories.form-control', {
+                onchange: ({ target }) => {
+                    model.newEorReason.category = target.value;
```

Comment:
As you are using everywhere `newEorReason` in this vnode, I would recommend you extract the variable in the begining.
`const {newEorReason} = model`
Moreover, you should limit the logic you apply in the view components. Perhaps it would be better to use the setter which also notifies. 
```
newEorReason = {category: target.value} 
```